nibabel/cifti2/cifti2.py

```diff
@@ -1490,6 +1503,33 @@ def get_data_dtype(self):
     def set_data_dtype(self, dtype):
         self._nifti_header.set_data_dtype(dtype)
 
+    def to_filename(self, filename, infer_intent=False):
+        """
+        Ensures NIfTI header intent code is set prior to saving.
+
+        Parameters
+        ----------
+        infer_intent : boolean, optional
+            If ``True``, attempt to infer and set intent code based on filename suffix.
+        """
+        header = self._nifti_header
+        if infer_intent:
```

Comment:
I think it would be reasonable to do something like:

```Python
predicted_intent = _infer_intent(...)
if infer_intent:
    set_intent()
else:
    if get_intent() != predicted_intent:
        warnings.warn(...)
```

Or we could make it `True`/`False`/`None`, where `True` overrides, `None` warns, and `False` suppresses warnings.